applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm

```diff
@@ -1,11 +1,102 @@
-<div class="cbi-section-create">
-	<% if self.invalid_cts then -%><div class="cbi-section-error"><% end %>
-	<input type="text" class="cbi-section-create-name" name="cbi.cts.<%=self.config%>.<%=self.sectiontype%>.text" />
-	<select class="cbi-section-create-name" name="cbi.cts.<%=self.config%>.<%=self.sectiontype%>.select">
-	<%- for k, v in luci.util.kspairs(self.add_select_options) do %>
-		<option value="<%=k%>"><%=luci.util.pcdata(v)%></option>
-	<% end -%>
-	</select>
-	<input class="cbi-button cbi-button-add" type="submit" value="<%:Add%>" title="<%:Add%>" />
-	<% if self.invalid_cts then %><br /><%:Invalid%></div><% end %>
+<script type="text/javascript">
+//<![CDATA[
+	function vpn_add()
+	{
+		var vpn_name     = div_add.querySelector("#instance_name1").value.replace(/[^\x00-\x7F]|[\s!@#$%^&*()+=\[\]{};':"\\|,<>\/?]/g,'');
+		var vpn_template = div_add.querySelector("#instance_template").value;
+		var form         = document.getElementsByName('cbi')[0];
+
+		if (!vpn_name || !vpn_name.length)
+		{
+			return info_message(vpn_output, "The 'Name' field must not be empty!", 2000);
```

Comment:
Consider marking these messages for translation:

`info_message(vpn_output, "<%=pcdata(translate("The 'Name' field must not be empty!"))%>", 2000);`

Reply:
done.